avro_derive/src/attributes/mod.rs

```diff
@@ -0,0 +1,206 @@
+                span,
+                "#[avro(rename_all = \"..\")] must match #[serde(rename_all = \"..\")], it's also deprecated. Please use only `#[serde(rename_all = \"..\")]`",
+            ))
+        }
+
+        if !errors.is_empty() {
+            return Err(errors);
+        }
+
+        Ok(Self {
+            name: avro.name,
+            namespace: avro.namespace,
+            doc: avro.doc,
+            alias: avro.alias,
+            rename_all: serde.rename_all.serialize,
+        })
+    }
+}
+
+pub struct FieldOptions {
+    pub doc: Option<String>,
+    pub default: Option<String>,
+    pub alias: Vec<String>,
+    pub rename: Option<String>,
+    pub skip: bool,
+    pub flatten: bool,
+}
+
+impl FieldOptions {
+    pub fn new(attributes: &[Attribute], span: Span) -> Result<Self, Vec<syn::Error>> {
+        let mut avro =
+            avro::FieldAttributes::from_attributes(attributes).map_err(darling_to_syn)?;
+        let mut serde =
+            serde::FieldAttributes::from_attributes(attributes).map_err(darling_to_syn)?;
+        // Sort the aliases, so our check for equality does not fail if they are provided in a different order
+        avro.alias.sort();
+        serde.alias.sort();
+
+        // Collect errors so user gets all feedback at once
+        let mut errors = Vec::new();
+
+        // Check for any Serde attributes that are hard errors
+        if serde.getter.is_some() {
+            errors.push(syn::Error::new(
+                span,
+                "AvroSchema derive does not support the Serde `getter` attribute",
+            ));
+        }
+
+        // Check for conflicts between Serde and Avro
+        if avro.skip && !serde.skip {
+            errors.push(syn::Error::new(
+                span,
+                "`#[avro(skip)]` requires `#[serde(skip)]`, it's also deprecated. Please use only `#[serde(skip)]`"
+            ));
+        }
+        if avro.flatten && !serde.flatten {
+            errors.push(syn::Error::new(
+                span,
+                "`#[avro(flatten)]` requires `#[serde(flatten)]`, it's also deprecated. Please use only `#[serde(flatten)]`"
+            ));
+        }
+        // TODO: rename and alias checking can be relaxed with a more complex check, would require the field name
+        if avro.rename.is_some() && serde.rename != avro.rename {
+            errors.push(syn::Error::new(
+                span,
+                "`#[avro(rename = \"..\")]` must match `#[serde(rename = \"..\")]`, it's also deprecated. Please use only `#[serde(rename  = \"..\")]`"
+            ));
+        }
+        if !avro.alias.is_empty() && serde.alias != avro.alias {
+            errors.push(syn::Error::new(
+                span,
+                "`#[avro(alias = \"..\")]` must match `#[serde(alias = \"..\")]`, it's also deprecated. Please use only `#[serde(alias  = \"..\")]`"
+            ));
+        }
+        if serde.skip_serializing && serde.skip_deserializing {
```

Comment:
Is this really an error that we (Avro) should break the build ?
It is up to the user whether to optimize the serde attributes or not.